exercises/list-ops/listops_test.go

```diff
@@ -0,0 +1,278 @@
+package listops
+
+import "reflect"
+import "testing"
+
+var foldTestCases = []struct {
+	name     string
+	property string
+	fn       binFunc
+	initial  int
+	list     IntList
+	want     int
+}{
+	{
+		name:     "empty list",
+		property: "foldl",
+		fn:       func(x, y int) int { return x * y },
+		initial:  2,
+		want:     2,
+		list:     []int{},
+	},
+	{
+		name:     "direction independent function applied to non-empty list",
+		property: "foldl",
+		fn:       func(x, y int) int { return x + y },
+		initial:  5,
+		want:     15,
+		list:     []int{1, 2, 3, 4},
+	},
+	{
+		name:     "direction dependent function applied to non-empty list",
+		property: "foldl",
+		fn:       func(x, y int) int { return x / y },
+		initial:  5,
+		want:     0,
+		list:     []int{2, 5},
+	},
+	{
+		name:     "empty list",
+		property: "foldr",
+		fn:       func(x, y int) int { return x * y },
+		initial:  2,
+		want:     2,
+		list:     []int{},
+	},
+	{
+		name:     "direction dependent function applied to non-empty list",
+		property: "foldr",
+		fn:       func(x, y int) int { return x + y },
+		initial:  5,
+		want:     15,
+		list:     []int{1, 2, 3, 4},
+	},
+	{
+		name:     "direction dependent function applied to non-empty list",
+		property: "foldr",
+		fn:       func(x, y int) int { return x / y },
+		initial:  5,
+		want:     2,
+		list:     []int{2, 5},
+	},
+}
+
+func TestFold(t *testing.T) {
+	var got int
+	for _, tt := range foldTestCases {
+		if tt.property == "foldr" {
+			got = tt.list.Foldr(tt.fn, tt.initial)
+		} else {
+			got = tt.list.Foldl(tt.fn, tt.initial)
+		}
+		if got != tt.want {
+			t.Fatalf("Build for test case %q for property %s returned %d but was expected to return %d",
+				tt.name, tt.property, got, tt.want)
+		}
```

Comment:
Perhaps
```
} else {
  t.Logf("PASS: %s", tt.name)
}
```
to allow 'go test -v' to show more output.  This has been typical in most exercises.  Ditto for the other test funcs.